context-propagation/src/main/java/io/micrometer/context/ContextAccessor.java

```diff
@@ -33,10 +33,10 @@
 public interface ContextAccessor<READ, WRITE> {
 
     /**
-     * Whether this accessor can capture values from the given type of context.
-     * @param contextType the type of external context
+     * {@link Class} representing the type of context this accessor is capable of
+     * reading values from.
      */
-    boolean canReadFrom(Class<?> contextType);
+    Class<?> readableType();
```

Comment:
I added the bounds. It might make accessors with the parameters being generic as well to be harder to implement and at times require raw types, as demonstrated in tests, but it should not be common for most implementations, where a concrete type will be specified.